app/controllers/concerns/wizard_steps.rb

```diff
@@ -42,6 +45,31 @@ def resend_verification
 
 private
 
+  def process_magic_link_token
+    token = request.query_parameters[:magic_link_token]
+    return if token.blank?
+
+    @wizard.process_magic_link_token(token)
+    redirect_to(step_path(@wizard.next_key(Wizard::Steps::Authenticate.key)))
+  rescue GetIntoTeachingApiClient::ApiError => e
+    handle_magic_link_token_error(e) && return if e.code == 401
+
+    raise
+  end
+
+  def handle_magic_link_token_error(error)
+    response = JSON.parse(error.response_body)
+    redirect_to(first_step_path(magic_link_token_error: response["status"]))
+  end
+
+  def display_magic_link_token_error
+    magic_link_token_error = params[:magic_link_token_error]
+    return if magic_link_token_error.blank?
+
+    message = t("activemodel.errors.magic_link_token.#{magic_link_token_error.underscore}")
```

Comment:
Yeah good catch, it would just show the translation key I believe. I'll update it to show a generic error as the API only returns a status indicator and not user-friendly errors 👍